include/powsybl/iidm/LoadingLimitsAdder.hxx

```diff
@@ -0,0 +1,168 @@
+/**
+ * Copyright (c) 2021, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#ifndef POWSYBL_IIDM_LOADINGLIMITSADDER_HXX
+#define POWSYBL_IIDM_LOADINGLIMITSADDER_HXX
+
+#include <powsybl/iidm/LoadingLimitsAdder.hpp>
+
+#include <unordered_set>
+
+#include <boost/range/adaptor/map.hpp>
+
+#include <powsybl/iidm/OperationalLimitsHolder.hpp>
+#include <powsybl/iidm/ValidationException.hpp>
+#include <powsybl/iidm/ValidationUtils.hpp>
+#include <powsybl/logging/Logger.hpp>
+#include <powsybl/logging/LoggerFactory.hpp>
+
+namespace powsybl {
+
+namespace iidm {
+
+template <typename L, typename A>
+LoadingLimitsAdder<L, A>::TemporaryLimitAdder::TemporaryLimitAdder(OperationalLimitsHolder& owner, LoadingLimitsAdder<L, A>& parent) :
+    m_owner(owner),
+    m_parent(parent) {
+}
+
+template <typename L, typename A>
+LoadingLimitsAdder<L, A>& LoadingLimitsAdder<L, A>::TemporaryLimitAdder::endTemporaryLimit() {
+    if (std::isnan(m_value)) {
+        throw ValidationException(m_owner, "temporary limit value is not set");
+    }
+    if (m_value <= 0) {
+        throw ValidationException(m_owner, "temporary limit value must be > 0");
+    }
+    if (!m_acceptableDuration) {
+        throw ValidationException(m_owner, "acceptable duration is not set");
+    }
+    checkAndGetUniqueName();
+    return m_parent.addTemporaryLimit(m_name, m_value, *m_acceptableDuration, m_fictitious);
+}
+
+template <typename L, typename A>
+void LoadingLimitsAdder<L, A>::TemporaryLimitAdder::checkAndGetUniqueName() {
+    if (m_name.empty()) {
+        throw ValidationException(m_owner, "name is not set");
+    }
+    if (m_ensureNameUnicity) {
+        unsigned long i = 0UL;
+        std::string uniqueName = m_name;
+        while (i < std::numeric_limits<unsigned long>::max() && nameExists(uniqueName)) {
+            uniqueName = m_name + stdcxx::format("#%1%", i);
+            i++;
+        }
+        m_name = uniqueName;
+    }
+}
+
+template <typename L, typename A>
+typename LoadingLimitsAdder<L, A>::TemporaryLimitAdder& LoadingLimitsAdder<L, A>::TemporaryLimitAdder::ensureNameUnicity() {
+    m_ensureNameUnicity = true;
+    return *this;
+}
+
+template <typename L, typename A>
+bool LoadingLimitsAdder<L, A>::TemporaryLimitAdder::nameExists(const std::string& name) const {
+    auto it = std::find_if(m_parent.m_temporaryLimits.begin(), m_parent.m_temporaryLimits.end(), [&name](const std::pair<unsigned long, LoadingLimits::TemporaryLimit>& item) {
+        return item.second.getName() == name;
+    });
+    return it != m_parent.m_temporaryLimits.end();
+}
```

Comment:
To be discussed: this method should be defined in parent class (`LoadingLimitsAdder)` and in child class (`TemporaryLimitAdder`), so that we call the first one from the second one? If yes, => I agree :)